File: backend/Models/Property.cs
namespace QBooking.Models
{
    public class Property
    {
        public int Id { get; set; }
        public int HostId { get; set; }
        public string Name { get; set; }
        public string Slug { get; set; }
        public int ProductTypeId { get; set; }
        public string Description { get; set; }
        public string ShortDescription { get; set; }
       public string AddressDetail { get; set; }
        public int? CommuneId { get; set; }
        public int ProvinceId { get; set; }
        public string PostalCode { get; set; }
        public decimal? Latitude { get; set; }
        public decimal? Longitude { get; set; }
        public byte StarRating { get; set; } = 0;
        public int TotalRooms { get; set; } = 0;
        public int? EstablishedYear { get; set; }
        public TimeSpan? CheckInTime { get; set; }
        public TimeSpan? CheckOutTime { get; set; }
        public int MinStayNights { get; set; } = 1;
        public int MaxStayNights { get; set; } = 30;
        public string CancellationPolicy { get; set; } = "moderate";
        public string MetaTitle { get; set; }
        public string MetaDescription { get; set; }
        public string MetaKeywords { get; set; }
        public decimal? PriceFrom { get; set; }
        public string Currency { get; set; } = "VND";
        public string Status { get; set; } = "draft";
        public bool IsActive { get; set; } = true;
        public bool IsFeatured { get; set; } = false;
        public int ViewCount { get; set; } = 0;
        public int BookingCount { get; set; } = 0;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;


        public User Host { get; set; }
        public Province Province { get; set; }
        public Commune Commune { get; set; }
        public ProductType ProductType { get; set; }

        public ICollection<PropertyImage> Images { get; set; } = new List<PropertyImage>();
        public ICollection<PropertyAmenity> Amenities { get; set; } = new List<PropertyAmenity>();
        public ICollection<RoomType> RoomTypes { get; set; } = new List<RoomType>();
        public ICollection<Booking> Bookings { get; set; } = new List<Booking>();
        public ICollection<Review> Reviews { get; set; } = new List<Review>();
        public ICollection<Favorite> Favorites { get; set; }

    }
}

File: frontend/src/types/main/property.ts
// types/property.ts

// Base API response structure
export interface ApiResponse<T = any> {
  success: boolean
  message: string
  statusCode: number
  data?: T
  error?: string
}

// Property response from API (matches your C# PropertyResponse)
export interface PropertyApiResponse {
  id: number
  name: string
  slug: string
  type: string
  description: string
  addressDetail: string
  province: string
  commune: string
  priceFrom: number
  currency: string
  status: string
  isFeatured: boolean
  isActive: boolean
  createdAt: string
  totalViews: number
  totalReviews: number
  averageRating: number | null
  images: string[]
  amenities: string[]
}

// Paginated response structure
export interface PaginatedPropertyResponse {
  properties: PropertyApiResponse[]
  totalCount: number
  page: number
  pageSize: number
  totalPages: number
}

// Updated filter request structure (matches your updated C# PropertyApprovedFilterRequest)
export interface PropertyFilterRequest {
  name?: string
  productTypeId?: number
  provinceId?: number
  amenityIds?: number[]
  minRating?: number
  // 🆕 New booking parameters
  checkIn?: string              // ISO date string (YYYY-MM-DD)
  checkOut?: string             // ISO date string (YYYY-MM-DD)
  adults?: number               // Number of adult guests
  children?: number             // Number of child guests
  page: number
  pageSize: number
}

// Product Type for filtering
export interface ProductType {
  id: number;
  name: string;
}

// Province for filtering
export interface Province {
  id: number
  name: string
  code?: string
}

// Amenity response structure
export interface AmenityResponse {
  id: number
  categoryId: number
  name: string
  slug: string
  icon: string
  description: string
  isPopular: boolean
  sortOrder: number
  categoryName: string
}

// Amenity category response structure
export interface AmenityCategoryResponse {
  id: number
  name: string
  slug: string
  icon: string
  description: string
  sortOrder: number
  amenities: AmenityResponse[]
}

// Converted Property for UI components
export interface Property {
  id: string
  title: string
  slug: string
  description: string
  location: {
    address: string
    city: string
    district?: string
    coordinates?: {
      lat: number
      lng: number
    }
  }
  price: {
    amount: number
    currency: string
    period: 'night' | 'month' | 'week'
  }
  images: string[]
  thumbnail: string
  rating: number
  reviewCount: number
  propertyType: 'apartment' | 'villa' | 'house' | 'hotel' | 'resort' | 'homestay'
  roomType: 'entire_place' | 'private_room' | 'shared_room'
  capacity: {
    guests: number
    bedrooms: number
    bathrooms: number
    beds: number
  }
  amenities: string[]
  features: string[]
  rules: string[]
  host: {
    id: string
    name: string
    avatar: string
    isVerified: boolean
    responseTime: string
    languages: string[]
  }
  availability: {
    isInstantBook: boolean
    minStay: number
    maxStay?: number
    checkIn: string
    checkOut: string
  }
  location_details: {
    neighborhood: string
    transportation: string[]
    nearby: string[]
  }
  cancellationPolicy: 'flexible' | 'moderate' | 'strict'
  isVerified: boolean
  isSuperhost: boolean
  createdAt: string
  updatedAt: string
  totalViews: number
}

// Updated property filters for UI filtering
export interface PropertyFilters {
  location?: string
  // 🆕 Updated booking parameters
  checkIn?: string              // ISO date string (YYYY-MM-DD)
  checkOut?: string             // ISO date string (YYYY-MM-DD)
  adults?: number               // Number of adult guests (default: 1)
  children?: number             // Number of child guests (default: 0)
  priceRange?: {
    min: number
    max: number
  }
  name?: string
  propertyType?: string[]
  roomType?: string[]
  amenities?: string[]           // Keep string array for backward compatibility
  amenityIds?: number[]          // New field for amenity IDs
  rating?: number                // This will map to minRating in API
  isInstantBook?: boolean
  sortBy?: 'price_low' | 'price_high' | 'rating' | 'newest' | 'popular'
  provinceId?: number
  productTypeId?: number
}

// Search parameters
export interface PropertySearchParams {
  query?: string
  page?: number
  limit?: number
  filters?: PropertyFilters
}


export interface PropertyForBooking {
  id: number
  name: string
  addressDetail: string
  province: string
  commune: string
  checkInTime: string
  checkOutTime: string
  starRating: number
  mainImage: string
  roomTypes: RoomTypeForBooking[]
}

export interface RoomTypeForBooking {
  id: number
  name: string
  maxAdults: number
  maxChildren: number
  maxGuests: number
  bedType: string
  basePrice: number
  roomImage: string
}